cpp/src/partitioning/partitioning.cu

```diff
@@ -775,11 +777,25 @@ std::pair<std::unique_ptr<table>, std::vector<size_type>> hash_partition(
   table_view const& input,
```

Comment:
Why not make this function templated?  Then you can remove the runtime failures and switch statement below, right?  That would let errors show up at compile-time instead of run-time, which is convenient for developers.

So for the caller, it would be:

```
hash_partition<IndentityHash>(input, columns_to_hash, num_paritions, steam, mr);
```

instead of:

```
hash_partition(input, columns_to_hash, num_paritions, hash_id::HASH_IDENTITY, steam, mr);
```
Also, you have written your check for data type twice, once here and once above with the `if_enable_t`.  That is repeating yourself.  Instead of doing the test twice, with this change, now you will only need to do it once.  Also this function will remain the same length instead of growing by 12 lines.

Reply:
It cannot be a template because most libcudf users are dynamic/interpreted languages like Python/Spark where the relevant information isn't known until runtime. Making the hash function a template parameter would just force the caller to do the switch statement.

Reply:
Okay!